src/js/letters/components/DownloadLetterLink.jsx

```diff
@@ -140,9 +90,18 @@ class DownloadLetterLink extends React.Component {
   }
 }
 
+function mapStateToProps(state, ownProps) {
+  return {
+    letterType: ownProps.letterType,
```

Comment:
I made this a connected component that still has its own props since we would rather get the option values from the store but pass in the letter type and name.